include/nanorange/detail/memory/temporary_vector.hpp

```diff
@@ -0,0 +1,94 @@
+
+// nanorange/detail/memory/temporary_vector.hpp
+//
+// Copyright (c) 2019 Tristan Brindle (tcbrindle at gmail dot com)
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef NANORANGE_DETAIL_MEMORY_TEMPORARY_VECTOR_HPP_INCLUDED
+#define NANORANGE_DETAIL_MEMORY_TEMPORARY_VECTOR_HPP_INCLUDED
+
+#include <nanorange/memory/destroy.hpp>
+
+#include <cassert>
+#include <memory>
+#include <new>
+
+NANO_BEGIN_NAMESPACE
+
+namespace detail {
+
+template <typename T>
+struct temporary_vector {
+private:
+    struct deleter {
+        void operator()(T* ptr) const { ::operator delete[](ptr); }
+    };
+
+public:
+    temporary_vector() = default;
+
+    explicit temporary_vector(std::size_t capacity)
+        : start_(static_cast<T*>(::operator new[](capacity * sizeof(T), std::nothrow))),
```

Comment:
Just to be annoying, I'm going to point out that this doesn't handle types with extended alignments - alignments greater than `alignof(max_align_t)`. You're perfectly welcome to not care.

Reply:
That did occur to me, but the "overloads" of `operator new` taking `std::align_val_t` are only available in C++17. Do you have any tips on how to handle this in C++14?

Reply:
In 14, your options are:
1. Ignore extended alignments and hope people don't use them.
2. `static_assert(alignof(T) <= alignof(std::max_align_t), "Extended alignments not supported.")`
3. Handle extended alignments manually by allocating `sizeof(void*) + alignof(T) - 1` additional bytes of space, using `std::align` to find the first address with the correct alignment, and storing a `void*` to the base of the allocated memory block after the "user" space to retrieve later when you want to free the memory.

3 is obviously way more work than 1 and 2.

Reply:
Thanks. I reckon at this point I'll just go for option 1 and see whether I ever get any bug reports. (I suspect I won't, unless you're feeling particularly evil one day!)